### ISSUE TITLE
ISO build failed, unmet dependencies: linux-kernel/linux-image

### ISSUE THREAD

**Issue by dd010101** [dd010101/vyos-jenkins#32]
All packages built successfully but still:

```
[2024-07-05 19:56:17] lb chroot_install-packages install
P: Begin installing packages (install pass)...
Reading package lists...
Building dependency tree...
Reading state information...
Package telegraf is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Unable to locate package linux-image-6.6.36-amd64-vyos
E: Couldn't find any package by glob 'linux-image-6.6.36-amd64-vyos'
E: Couldn't find any package by regex 'linux-image-6.6.36-amd64-vyos'
E: Unable to locate package vyos-linux-firmware
E: Unable to locate package vyos-intel-qat
E: Unable to locate package vyos-intel-ixgbe
E: Unable to locate package vyos-intel-ixgbevf
E: Unable to locate package openvpn-dco
E: Package 'telegraf' has no installation candidate
E: An unexpected failure occurred, exiting...
P: Begin unmounting filesystems...
P: Saving caches...
Reading package lists...
Building dependency tree...
Reading state information...
I: Checking if packages required for VyOS image build are installed
I: using build flavors directory data/build-flavors
I: Cleaning the build workspace
I: Setting up additional APT entries
I: Configuring live-build
I: Starting image build
Traceback (most recent call last):
  File "/vyos/./build-vyos-image", line 621, in <module>
    cmd("lb build 2>&1")
  File "/vyos/scripts/image-build/utils.py", line 84, in cmd
    raise OSError(f"Command '{command}' failed")
OSError: Command 'lb build 2>&1' failed
ISO build failed
```

_Originally posted by @koljenovic in https://github.com/dd010101/vyos-jenkins/issues/19#issuecomment-2211359939_

**Comment by dd010101** [comment]
@koljenovic

Did you look at the **Build history**, follow the [**Something is wrong**](https://github.com/dd010101/vyos-jenkins?tab=readme-ov-file#something-is-wrong) and look for the packages **linux-kernel** and provide mentioned logs.

**Comment by koljenovic** [comment]
I have built all the packages by hand prior to finding your repo builder and had a similar issue:

P.S. Will look into build history further and report.

```
[2024-07-05 13:40:15] lb chroot_install-packages install
P: Begin installing packages (install pass)...
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package linux-image-6.6.36-amd64-vyos
E: Couldn't find any package by glob 'linux-image-6.6.36-amd64-vyos'
E: Couldn't find any package by regex 'linux-image-6.6.36-amd64-vyos'
E: Unable to locate package vyos-xe-guest-utilities
E: Unable to locate package vyos-linux-firmware
E: Unable to locate package vyos-intel-qat
E: Unable to locate package vyos-intel-ixgbe
E: Unable to locate package vyos-intel-ixgbevf
E: Unable to locate package openvpn-dco
E: Unable to locate package telegraf
E: Unable to locate package vyos-1x-smoketest
E: Unable to locate package vyos-1x-vmware
E: Unable to locate package vyos-world
E: Unable to locate package vyos-user-utils
E: An unexpected failure occurred, exiting...
P: Begin unmounting filesystems...
P: Saving caches...
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Traceback (most recent call last):
  File "/vyos/./build-vyos-image", line 621, in <module>
    cmd("lb build 2>&1")
  File "/vyos/scripts/image-build/utils.py", line 84, in cmd
    raise OSError(f"Command '{command}' failed")
OSError: Command 'lb build 2>&1' failed
```

**Comment by koljenovic** [comment]
Here are all the deb packages I have:

```
/var/lib/jenkins/VyOS/sagitta/openvpn-dco_0.2.20231117_amd64.deb
/var/lib/jenkins/VyOS/sagitta/librtr-doc_0.8.0_all.deb
/var/lib/jenkins/VyOS/sagitta/nftables_1.0.9-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac5-bin-dbgsym_1.7.0-0.1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libcharon-extauth-plugins-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/eapoltest-dbgsym_2.10-1028-g6b9c86466_amd64.deb
/var/lib/jenkins/VyOS/sagitta/twamp-client-dbgsym_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/keepalived_2.2.8-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-dev_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-relay_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac2_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libstrongswan-standard-plugins-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libyang2-tools_2.1.148-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/accel-ppp.deb
/var/lib/jenkins/VyOS/sagitta/linux-headers-6.6.36-amd64-vyos_6.6.36-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/librtr-dbg_0.8.0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-swanctl-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libstrongswan-extra-plugins-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-nm_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/python3-nftables_1.0.9-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libcharon-extauth-plugins_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/rtr-tools_0.8.0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-cfg-system_0.20.44+vyos2+current22_amd64.deb
/var/lib/jenkins/VyOS/sagitta/dropbear-bin-dbgsym_2022.83-1+deb12u1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/keepalived-dbgsym_2.2.8-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnftnl11_1.2.6-2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/wide-dhcpv6-server-dbgsym_20080615-23_amd64.deb
/var/lib/jenkins/VyOS/sagitta/ndppd-dbgsym_0.2.5-6_amd64.deb
/var/lib/jenkins/VyOS/sagitta/frr-dbgsym_9.1.1-18-gbadb6f1fc_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac5-bin_1.7.0-0.1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/owamp-client_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/wide-dhcpv6-server_20080615-23_amd64.deb
/var/lib/jenkins/VyOS/sagitta/charon-systemd-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-charon-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/python3-vici_5.9.11-1_all.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-bash_4.1-3+vyos2+current2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/frr-doc_9.1.1-18-gbadb6f1fc_all.deb
/var/lib/jenkins/VyOS/sagitta/libcharon-extra-plugins-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/openvpn-otp_1.0-4-g47f8ccf_amd64.deb
/var/lib/jenkins/VyOS/sagitta/nftables-dbgsym_1.0.9-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-1x_1.4.0-78-g6074432a1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/librtr0_0.8.0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/owamp-server_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-1x-smoketest_1.4.0-78-g6074432a1_all.deb
/var/lib/jenkins/VyOS/sagitta/libpam-tacplus-dbgsym_1.7.0-0.1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/twamp-server-dbgsym_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-world_1.4.0+vyos1+current_all.deb
/var/lib/jenkins/VyOS/sagitta/rtr-tools-dbg_0.8.0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libpam-tacplus_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/dropbear-initramfs_2022.83-1+deb12u1_all.deb
/var/lib/jenkins/VyOS/sagitta/frr-rpki-rtrlib_9.1.1-18-gbadb6f1fc_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-wanloadbalance-dbgsym_0.13.70+vyos2+current1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/eapoltest_2.10-1028-g6b9c86466_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-libcharon_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnftnl-dev-doc_1.2.6-2_all.deb
/var/lib/jenkins/VyOS/sagitta/libyang-tools_2.1.148-1_all.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-bash-dbgsym_4.1-3+vyos2+current2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/telegraf_1.28.3-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/frr-pythontools_9.1.1-18-gbadb6f1fc_all.deb
/var/lib/jenkins/VyOS/sagitta/libyang2_2.1.148-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/hostapd_2.10-1028-g6b9c86466_amd64.deb
/var/lib/jenkins/VyOS/sagitta/charon-cmd_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/accel-ppp_1.12.0-260-g19c36e5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-biosdevname_0.3.11+vyos2+current2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac2-dbgsym_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libpam-radius-auth-dbgsym_1.5.0-cl3u7_amd64.deb
/var/lib/jenkins/VyOS/sagitta/owamp-server-dbgsym_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnftnl-dev_1.2.6-2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libvyatta-cfg1_0.102.0+vyos2+current5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libyang2-dev_2.1.148-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-swanctl_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-charon_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/pmacct_1.7.7-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-utils_0.0.3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-keama_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/opennhrp_0.14-20-g613277f_amd64.deb
/var/lib/jenkins/VyOS/sagitta/ipaddrcheck_1.2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-server-ldap_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libstrongswan_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac2-bin-dbgsym_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-common_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/ddclient_3.11.2+vyos0_all.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-libcharon-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-pki_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-server-dbgsym_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnftnl11-dbgsym_1.2.6-2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/charon-systemd_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-biosdevname-dbgsym_0.3.11+vyos2+current2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/dropbear-run_2022.83-1+deb12u1_all.deb
/var/lib/jenkins/VyOS/sagitta/wpasupplicant_2.10-1028-g6b9c86466_amd64.deb
/var/lib/jenkins/VyOS/sagitta/frr_9.1.1-18-gbadb6f1fc_amd64.deb
/var/lib/jenkins/VyOS/sagitta/wpasupplicant-dbgsym_2.10-1028-g6b9c86466_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libvyosconfig0-dbgsym_0.0.10_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-utils-dbgsym_0.0.3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/cloud-init_22.1-444-g61d7ba9b-1~bddeb_all.deb
/var/lib/jenkins/VyOS/sagitta/jool_4.1.9+bf4c7e3669-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/ipaddrcheck-dbgsym_1.2_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac5_1.7.0-0.1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac5-dbgsym_1.7.0-0.1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/dropbear-bin_2022.83-1+deb12u1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/dropbear_2022.83-1+deb12u1_all.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-cfg-dbgsym_0.102.0+vyos2+current5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtacplus-map1_1.0.1-cl5.1.0u9_amd64.deb
/var/lib/jenkins/VyOS/sagitta/live-boot-initramfs-tools_20151213-vyos0_all.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-cfg_0.102.0+vyos2+current5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/aws-gwlbtun.deb
/var/lib/jenkins/VyOS/sagitta/librtr-dev_0.8.0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/radvd_2.20-rc1-23-gf2de476_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-op_0.14.0+vyos2+current8_all.deb
/var/lib/jenkins/VyOS/sagitta/strongswan_5.9.11-2+vyos0_all.deb
/var/lib/jenkins/VyOS/sagitta/ethtool-dbgsym_6.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-1x-dbgsym_1.4.0-78-g6074432a1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-server_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-user-utils_1.4.0+vyos1+current_all.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-relay-dbgsym_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libvyatta-cfg1-dbgsym_0.102.0+vyos2+current5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/python3-pyhumps_3.8.0-1_all.deb
/var/lib/jenkins/VyOS/sagitta/libpam-radius-auth_1.5.0-cl3u7_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-1x-vmware_1.4.0-78-g6074432a1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libwpa-client-dev_2.10-1028-g6b9c86466_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-intel-ixgbevf_4.18.9_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libvyosconfig0_0.0.10_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac2-bin_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnftables1_1.0.9-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/radius-shell-dbgsym_1.5.0-cl3u7_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnftables1-dbgsym_1.0.9-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-client_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac-dev_1.7.0-0.1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnftables-dev_1.0.9-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/wide-dhcpv6-client-dbgsym_20080615-23_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libstrongswan-standard-plugins_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/hvinfo-dbgsym_1.2.0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-client-ddns_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libyang2-dbgsym_2.1.148-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/wide-dhcpv6-relay-dbgsym_20080615-23_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libpam-tacplus_1.7.0-0.1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-starter-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/twamp-client_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/pmacct-dbgsym_1.7.7-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/hsflowd_2.0.55-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/jool-dbgsym_4.1.9+bf4c7e3669-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libyang2-tools-dbgsym_2.1.148-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtac-dev_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/twamp-server_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-keama-dbgsym_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/hvinfo_1.2.0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/live-boot-doc_20151213-vyos0_all.deb
/var/lib/jenkins/VyOS/sagitta/libnss-mapuser_1.1.0-cl3u3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-client-ddns-dbgsym_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/charon-cmd-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/linux-libc-dev_6.6.36-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-intel-ixgbe_5.20.3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnss-tacplus_1.0.4-cl5.1.0u11_amd64.deb
/var/lib/jenkins/VyOS/sagitta/cloud-init_all.deb
/var/lib/jenkins/VyOS/sagitta/vyos-http-api-tools-dbgsym_2.3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-client-dbgsym_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/frr-snmp_9.1.1-18-gbadb6f1fc_amd64.deb
/var/lib/jenkins/VyOS/sagitta/ndppd_0.2.5-6_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-xe-guest-utilities_7.13.0+vyos1.3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnss-tacplus-dbgsym_1.0.4-cl5.1.0u11_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libstrongswan-extra-plugins_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyatta-wanloadbalance_0.13.70+vyos2+current1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtacplus-map-dev_1.0.1-cl5.1.0u9_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libvyatta-cfg-dev_0.102.0+vyos2+current5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/udp-broadcast-relay_0.1+vyos3+equuleus1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/wide-dhcpv6-relay_20080615-23_amd64.deb
/var/lib/jenkins/VyOS/sagitta/radius-shell_1.5.0-cl3u7_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libnss-mapuser-dbgsym_1.1.0-cl3u3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-pki-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/strongswan-starter_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/frr-rpki-rtrlib-dbgsym_9.1.1-18-gbadb6f1fc_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-http-api-tools_2.3_amd64.deb
/var/lib/jenkins/VyOS/sagitta/live-boot_20151213-vyos0_all.deb
/var/lib/jenkins/VyOS/sagitta/wide-dhcpv6-client_20080615-23_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libpam-tacplus-dbgsym_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/owamp-client-dbgsym_4.4.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-intel-qat_4.24.0-00005-0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libcharon-extra-plugins_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/ethtool_6.6-1_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libtacplus-map1-dbgsym_1.0.1-cl5.1.0u9_amd64.deb
/var/lib/jenkins/VyOS/sagitta/isc-dhcp-server-ldap-dbgsym_4.4.3-P1-4_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libpam-tacplus-dev_1.4.3-cl5.1.0u5_amd64.deb
/var/lib/jenkins/VyOS/sagitta/libstrongswan-dbgsym_5.9.11-2+vyos0_amd64.deb
/var/lib/jenkins/VyOS/sagitta/vyos-linux-firmware_20231211_all.deb
/var/lib/jenkins/VyOS/sagitta/hostapd-dbgsym_2.10-1028-g6b9c86466_amd64.deb
/var/lib/jenkins/VyOS/sagitta/frr-snmp-dbgsym_9.1.1-18-gbadb6f1fc_amd64.deb
/var/lib/jenkins/VyOS/sagitta/linux-image-6.6.36-amd64-vyos_6.6.36-1_amd64.deb
```

**Comment by koljenovic** [comment]
Attached a build log for linux-kernel.
[linux-kernel.log](https://github.com/user-attachments/files/16113535/linux-kernel.log)

**Comment by dd010101** [comment]
What do you see in the APT repository, try to:

```
curl http://172.17.17.17/sagitta/dists/sagitta/main/binary-amd64/Packages
```

What it returns?

**Comment by koljenovic** [comment]
linux-kernel nowhere to be seen ...
[Uploading Packages.txt…]()

**Comment by dd010101** [comment]
Then please follow the [**Something is wrong**](https://github.com/dd010101/vyos-jenkins?tab=readme-ov-file#something-is-wrong) and look into the uncron log, the Jenkins does it's job but the result doesn't get into the repository, thus uncron is the suspect.

**Comment by koljenovic** [comment]
Will do, thank you.

**Comment by dd010101** [comment]
You don't see linux-image in the uncron log?

```
journalctl --no-pager -b -u uncron.service, 
```

It would be nice to know the cause.

**Comment by koljenovic** [comment]
It shows up in the uncron logs, but it's not listed in the Packages and not present in the /pool/main:

`Jul 05 21:44:33 debian-builder uncron[889]: uncron: [INFO] Job "reprepro -v -b /home/sentrium/web/dev.packages.vyos.net/public_html/repositories/sagitta -A amd64 includedeb sagitta ~/VyOS/sagitta/linux-image-6.6.36-amd64-vyos_6.6.36-1_amd64.deb" was added to the queue`

Build for the following packages has faild in the first run due to low disk space and was cancelled and rerun manually after resizing the disk and the jobs ran successfully and produced a deb:

```
vyatta-wanloadbalance
vyatta-bash
telegraf
strongswan
netfilter
hostap
frr
linux-kernel
wide-dhcpv6
```

Afterwards I re-run steps from step 4 to try and resolve the issue to no avail.

**Comment by dd010101** [comment]
The uncron message `was added to the queue` tells you that the package was built and was handed over to the uncron and so sometimes later the uncron should process this package - the processing should follow after the `was added to the queue`. Do you see any later mentions of the linux-image? This is likely where the error is.

If your system run out of disk space then there can be number of uncompleted tasks, so you can destroy the VM and rerun everything if that's easier for you.

**Comment by koljenovic** [comment]
Since running out of disk space during a VyOS build process is a fringe error condition I would rather not debug any further. I will rerun the build with a new clean debian VM.

Just to note that 40 GB VM is not enough to complete the build.

I will report if the new clean build passes.

**Comment by koljenovic** [comment]
```
ISO build is complete.
The file is called: vyos-1.4.0-release-20240706-iso-amd64.iso.
```

**Comment by dd010101** [comment]
👍 Looks good! Once you get ISO the likeliness of success is very high, but you shall test it by actual usage.

What disk size you did end up using?

**Comment by koljenovic** [comment]
Running smoketest rn, I ended up going with 90 GB this time, but I believe 50 GB should be more than enough for a single branch build (e.g. sagitta). First time I built all the branches (forgot to specify env var after reboot) and it filled up 40 GB easy.

Possible UI improvement would be to display and persist a branch selector in [1-prereqs.sh](https://github.com/dd010101/vyos-jenkins/blob/master/1-prereqs.sh).
